lib/options.go

```diff
@@ -572,6 +575,11 @@ func (o Options) Validate() []error {
 					o.ExecutionSegment, o.ExecutionSegmentSequence))
 		}
 	}
+	for _, ip := range o.BlacklistIPs {
+		if ip.String() == "<nil>" {
+			errors = append(errors, fmt.Errorf("blacklisted IP %q has an invalid CIDR", ip.IP.String()))
+		}
```

Comment:
This is a very strange check, why do you compare against the string? And the error message doesn't really follow logically from the check. And in the check you're comparing `ip`, but in the error message you're referencing `ip.IP` :confused: 

So this is weird in multiple different ways, and if needs to be this way for some reason, that reason should be mentioned in a code comment here.

Reply:
And probably in the PR description, since this seems like a different fix than just being able to marshal the `blacklistIPs` in the JSON properly? :confused: